src/lib.rs

```diff
@@ -219,6 +282,55 @@ pub fn link(obj_file: &Path, output: &Path) -> Result<(), io::Error> {
     }
 }
 
+/// Structure used to handle JITing C code.
+///
+/// You should use `RccJIT::from_module` or `RccJIT::compile_string` to create instance of RccJIT.
+pub struct RccJIT {
+    module: Module<SimpleJITBackend>,
+}
+impl RccJIT {
+    /// Instantiate RccJIT from SimpleJIT Module.
+    pub fn from_module(m: Module<SimpleJITBackend>) -> Self {
+        Self { module: m }
+    }
+    /// Compile string and return JITed code.
+    pub fn compile_string(
+        program: &str,
+        opt: &Opt,
+    ) -> (Result<Self, Error>, VecDeque<CompileWarning>) {
+        let module = initialize_jit_module();
+        let (result, warnings) = compile::<SimpleJITBackend>(module, program, opt);
+        let result = result.map(|x| RccJIT::from_module(x));
+        (result, warnings)
+    }
+
+    /// Invoke this function before trying to get access to "new" compiled functions.
+    pub fn finalize(&mut self) {
+        self.module.finalize_definitions();
+    }
+    /// Get compiled function, if this function doesn't exist then `None` is returned, otherwise it's address returned.
+    ///
+    /// # Panics
+    /// Panics if function is not compiled (finalized). Try to invoke `finalize` before using `get_compiled_function`.
+    pub fn get_compiled_function(&mut self, name: &str) -> Option<*const u8> {
```

Comment:
Is there a reason you didn't call `finalize()` here instead of having the user do it?

Reply:
> Is there a reason you didn't call `finalize()` here instead of having the user do it?

There are no function in `Module` that may say is module finalized or no, and users may not need to finalize all functions in module.